src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2450,7 +2450,21 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatement(SyntaxKind par
                     }
 
                     var incompleteMember = _syntaxFactory.IncompleteMember(attributes, modifiers.ToList(), type.IsMissing ? null : type);
-                    if (incompleteMember.ContainsDiagnostics)
+
+                    var containsErrorDiagnostic = false;
+                    foreach (var child in incompleteMember.EnumerateNodes())
```

Comment:
ContainsDiagnostics returns true for warning diagnostics which is not what we want.